main/src/ca/uwaterloo/flix/language/dbg/prettierPrettyPrinting/AstPrinter.scala

```diff
@@ -0,0 +1,51 @@
+package ca.uwaterloo.flix.language.dbg.prettierPrettyPrinting
+
+import ca.uwaterloo.flix.api.Flix
+import ca.uwaterloo.flix.language.ast.SourceLocation
+import ca.uwaterloo.flix.language.dbg.prettierPrettyPrinting.Doc.Indent
+import ca.uwaterloo.flix.util.InternalCompilerException
+
+import java.nio.file.{Files, LinkOption, Path}
+
+object AstPrinter {
+
+  /**
+    * The extension of intermediate flix files.
+    */
+  val IREXTENSION: String = "flixir"
+
+  /**
+    * Prints `ast` to `build/ast/<phase>.<AstPrinter.IREXTENSION>` if
+    * `flix.options.xprintasts` contains `phase`.
+    *
+    * @param width  the maximum width of the output, default 80
+    * @param indent the indentation width of each indentation level, default 4
+    */
+  def printAst(phase: => String, ast: => DocAst.Program, width: Int = 80, indent: Int = 4)(implicit flix: Flix): Unit = {
```

Comment:
I think we should have these defaults in exactly one place. I see somewhere else you've also written 4 for indent, but IMO these should be a constant somewhere.